lib/rules/no-promise-executor-return.js

```diff
@@ -59,6 +60,58 @@ function isPromiseExecutor(node, scope) {
         isGlobalReference(parent.callee, getOuterScope(scope));
 }
 
+/**
+ * Checks if the given node is a void expression.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} - `true` if the node is a void expression
+ */
+function expressionIsVoid(node) {
+    return node.type === "UnaryExpression" && node.operator === "void";
+}
+
+/**
+ * Fixes the linting error by prepending "void " to the given node
+ * @param {ASTNode} node The node to fix.
+ * @param {Object} fixer The fixer object provided by ESLint.
+ * @returns {Array<Object>|Object} - An array of fix objects or fix to apply to the node.
+ */
+function voidPrependFixer(node, fixer) {
+
+    /*
+     * prepending `void ` will fail if
+     * the expression is () => void () => {}
+     * therefore, check if the expression is a function
+     */
+    if (node.type === "ArrowFunctionExpression") {
```

Comment:
another sharp corner, thanks for pointing out

Reply:
Also, if the expression is already in parentheses, it might make sense to just insert `void ` before the left parenthesis.

For example:

```js
new Promise(r => (foo ? r(1): r(2)));
```

could be fixed to:

```js
new Promise(r => void (foo ? r(1): r(2)));
```

Reply:
This "avoid adding parentheses if there are already" can be applied to previous implementation of "add () around arrow function expression before prepending void" logic

would you like me to do this redundant parentheses check for the arrow function as well?

Reply:
nevermind, `if (astUtils.getPrecedence(node) < astUtils.getPrecedence({ type: "UnaryExpression", operator: "void" })) {` should take care of that